sunpy/physics/differential_rotation.py

```diff
@@ -227,102 +407,367 @@ def _warp_sun_coordinates(xy, smap, dt: u.s, **diffrot_kwargs):
+    performed by calculating the distance of every pixel from the center of
+    the Sun.  If they are all off-disk, then the function returns True.
+    Otherwise, the function returns False.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map`
+        The input map.
+
+    Returns
+    -------
+    is_all_off_disk : `~bool`
+        Returns True if all map pixels are strictly more than one solar radius
+        away from the center of the Sun.
+    """
+    return np.all(~coordinate_is_on_disk(all_coordinates_from_map(smap), smap.rsun_obs))
+
+
+def is_all_on_disk(smap):
+    """
+    Checks if the entire map is on the solar disk.  The check is
+    performed by calculating the distance of every pixel from the center of
+    the Sun.  If they are all on-disk, then the function returns True.
+    Otherwise, the function returns False.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map`
+        The input map.
+
+    Returns
+    -------
+    is_all_on_disk : `~bool`
+        Returns True if all map pixels are strictly less than one solar radius
+        away from the center of the Sun.
+    """
+    return np.all(coordinate_is_on_disk(all_coordinates_from_map(smap), smap.rsun_obs))
+
+
+def contains_limb(smap):
+    """
+    Checks if a map contains a portion of the solar limb.  The check is
+    performed by calculating the distance of every pixel from the center of
+    the Sun.  If at least one pixel is on disk and at least one pixel is off
+    disk, the function returns True.  Otherwise, the function returns False.
+    Note that this function will also true if the entire solar limb is within
+    the field of view of the map.  Note also that in the case of coronagraph
+    images the limb itself need not be observed.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map`
+        The input map.
+
+    Returns
+    -------
+    contains_limb : `~bool`
+        Returns True If at least one pixel is on disk and at least one pixel
+        is off disk.
+    """
+    on_disk = coordinate_is_on_disk(all_coordinates_from_map(smap), smap.rsun_obs)
+    return np.logical_and(np.any(on_disk), np.any(~on_disk))
+
+
+@u.quantity_input
+def coordinate_is_on_disk(coordinate, scale: u.arcsecond):
+    """
+    Checks if the coordinate is on disk.  The check is performed by
+    comparing converting the co-ordinate to Helioprojective co-ordinates
+    and then comparing the coordinate's distance from the center of
+    the Sun to the solar radius.
+
+    Parameters
+    ----------
+    coordinate : `~sunpy.coordinate`
+        The input coordinate.
+
+    scale : `~astropy.units`
+        The pixel scale size in units of arcseconds.
+
+    Returns
+    -------
+    is_on_disk : `~bool`
+        Returns True if the coordinate is on disk, False otherwise.
+    """
+    # Calculate the radii of every pixel in helioprojective Cartesian
+    # co-ordinate distance units.
+    c = coordinate.transform_to(Helioprojective)
```

Comment:
Yes, you're right.  Maybe the check itself should change.  You want to make sure that the coordinate is on the disk as seen by the observer.  An observer property should also be passed in?

Reply:
Actually maybe this function should only accept Helioprojective coordinates so that it is left to the user to ensure that the correct co-ordinates are passed in.

Reply:
I think that is probably the best the best option. I think that `isinstance(c.frame, frames.Helioprojective)` should work.

Reply:
This needs fixing.